plugin/storage/cassandra/schema/schema.go

```diff
@@ -0,0 +1,200 @@
+		}
+
+		cfg.CompactionWindowSize, err = strconv.Atoi(cfg.CompactionWindow[:len(cfg.CompactionWindow)-1])
+		if err != nil {
+			return err
+		}
+
+		cfg.CompactionWindowUnit = cfg.CompactionWindow[len(cfg.CompactionWindow)-1:]
+	} else {
+		traceTTLMinutes := cfg.TraceTTL / 60
+
+		cfg.CompactionWindowSize = (traceTTLMinutes + 30 - 1) / 30
+		cfg.CompactionWindowUnit = "m"
+	}
+
+	switch cfg.CompactionWindowUnit {
+	case `m`:
+		cfg.CompactionWindowUnit = `MINUTES`
+	case `h`:
+		cfg.CompactionWindowUnit = `HOURS`
+	case `d`:
+		cfg.CompactionWindowUnit = `DAYS`
+	default:
+		return fmt.Errorf("Invalid compaction window unit. If can be among {m|h|d}")
+	}
+
+	return nil
+}
+
+func getQueryFileAsBytes(fileName string, cfg *config.Schema) ([]byte, error) {
+
+	tmpl, err := template.ParseFS(schemaFile, fileName)
+	if err != nil {
+		return nil, err
+	}
+
+	var result bytes.Buffer
+	err = tmpl.Execute(&result, cfg)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return result.Bytes(), nil
+}
+
+func getQueriesFromBytes(queryFile []byte) ([]string, error) {
+	lines := bytes.Split(queryFile, []byte("\n"))
+
+	var extractedLines [][]byte
+
+	for _, line := range lines {
+		// Remove any comments, if at the end of the line
+		commentIndex := bytes.Index(line, []byte(`--`))
+		if commentIndex != -1 {
+			// remove everything after comment
+			line = line[0:commentIndex]
+		}
+
+		if len(line) == 0 {
+			continue
+		}
+
+		extractedLines = append(extractedLines, bytes.TrimSpace(line))
+	}
+
+	var queries []string
+
+	// Construct individual queries strings
+	var queryString string
+	for _, line := range extractedLines {
+		queryString += string(line)
+		if bytes.HasSuffix(line, []byte(";")) {
+			queries = append(queries, queryString)
+			queryString = ""
+		}
+	}
+
+	if len(queryString) > 0 {
+		return nil, fmt.Errorf(`Invalid template`)
+	}
+
+	return queries, nil
+}
+
+func getCassandraQueriesFromQueryStrings(session cassandra.Session, queries []string) []cassandra.Query {
+	var casQueries []cassandra.Query
+
+	for _, query := range queries {
+		casQueries = append(casQueries, session.Query(query))
+	}
+
+	return casQueries
+}
+
+func contructSchemaQueries(session cassandra.Session, cfg *config.Schema) ([]cassandra.Query, error) {
+	err := constructCompleteSchemaConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	queryFile, err := getQueryFileAsBytes(`v004-go-tmpl.cql.tmpl`, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	queryStrings, err := getQueriesFromBytes(queryFile)
+	if err != nil {
+		return nil, err
+	}
+
+	casQueries := getCassandraQueriesFromQueryStrings(session, queryStrings)
+
+	return casQueries, nil
+}
+
+func GenerateSchemaIfNotPresent(session cassandra.Session, cfg *config.Schema) error {
+	casQueries, err := contructSchemaQueries(session, cfg)
+	if err != nil {
+		return err
+	}
+
+	for _, query := range casQueries {
+		err := query.Exec()
```

Comment:
If a corresponding table already exists, will this be a no-op or will it update the settings in the table? We need to allow users to be able to customize the settings.

Reply:
All the table creation queries use: `CREATE TABLE IF NOT EXISTS`, so this will be a no-op.